src/main/java/com/xebialabs/overthere/util/WindowsCommandLineArgsSanitizer.java

```diff
@@ -24,9 +24,18 @@
 
 public class WindowsCommandLineArgsSanitizer {
 
-    private static final char[] SPECIAL_CHARS = "|<>&^\r\n".toCharArray();
+    private static final char[] SPECIAL_CHARS = " |<>&^\r\n".toCharArray();
```

Comment:
But how is that solved now? Because you don't escape anything if there's a space. Isn't a space escapable using a caret?

Op 27 jun. 2016 11:26 a.m. schreef Sameer Arora <notifications@github.com>:

In src/main/java/com/xebialabs/overthere/util/WindowsCommandLineArgsSanitizer.java<https://github.com/xebialabs/overthere/pull/180#discussion_r68546485>:

> @@ -24,9 +24,18 @@
>
>  public class WindowsCommandLineArgsSanitizer {
>
> -    private static final char[] SPECIAL_CHARS = "|<>&^\r\n".toCharArray();
> +    private static final char[] SPECIAL_CHARS = " |<>&^\r\n".toCharArray();


No, it breaks in certain scenarios like a winrs -r command run from a localhost as winrs proxy. This one is a specific case.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/xebialabs/overthere/pull/180/files/7f37006777236fff80a89b3317c9162ef5028fc8#r68546485>, or mute the thread<https://github.com/notifications/unsubscribe/AAHLo-gwdBPH-1T58SQWob6pwYDCeXutks5qP5dJgaJpZM4I-Sv->.